### PR TITLE
chore: bump outdated dependencies

### PR DESCRIPTION
**Pull request by ckerr**
1. bumps the `standard` dependency from 8 to 12. Add `chai-dirty`, and update tests to use it, to avoid spurious warnings in BDD-style tests.

2. bump all the dependencies shown in `npm outdated`

3. confirm that `npm run generate` and `npm test` pass

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,32 +5,33 @@
   "author": "Zeke Sikelianos <zeke@sikelianos.com> (http://zeke.sikelianos.com)",
   "bin": "cli.js",
   "dependencies": {
-    "cheerio": "^0.22.0",
+    "cheerio": "^1.0.0-rc.2",
```
[the remark this answers is not in the document]
Comment (ckerr) [reply package.json]: I agree with this idea in general, but not in the case of cheerio: its latest stable release came out over two years ago and gets dinged in the audit warnings. On balance I think it's better to go with the release candidate.

File: package.json [hunk 2]
```diff
@@ -5,32 +5,33 @@
...
     "clean-deep": "^2.0.1",
-    "decamelize": "^1.2.0",
-    "dedent": "^0.6.0",
-    "electron-docs": "^2.0.1",
-    "entities": "^1.1.1",
+    "decamelize": "^2.0.0",
+    "dedent": "^0.7.0",
+    "electron-docs": "^3.0.2",
```
[the remark this answers is not in the document]
Comment (ckerr) [reply package.json]: Confirmed that the `electron.json` files generated by `master` and by this branch are identical

[the remark this answers is not in the document]
Comment (ckerr) [reply package.json]: @MarshallOfSound done

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (codebytere) [reply package.json]: I'd agree with the above, not as a rule but in this instance given the long lag window in stable releases

**Pull request merged**